modules/slugs/includes/qtranslate-slug-settings.php

```diff
@@ -486,109 +316,10 @@ function qts_validate_options( $input ) {
-                            'dl'         => array(),
-                            'dt'         => array(),
-                            'em'         => array(),
-                            'i'          => array(),
-                            'li'         => array(),
-                            'ol'         => array(),
-                            'p'          => array(),
-                            'q'          => array( 'cite' => array() ),
-                            'strong'     => array(),
-                            'ul'         => array(),
-                            'h1'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            ),
-                            'h2'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            ),
-                            'h3'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            ),
-                            'h4'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            ),
-                            'h5'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            ),
-                            'h6'         => array(
-                                'align' => array(),
-                                'class' => array(),
-                                'id'    => array(),
-                                'style' => array()
-                            )
-                        );
-
-                        $input[ $option['id'] ]       = trim( $input[ $option['id'] ] ); // trim whitespace
-                        $input[ $option['id'] ]       = force_balance_tags( $input[ $option['id'] ] ); // find incorrectly nested or missing closing tags and fix markup
-                        $input[ $option['id'] ]       = wp_kses( $input[ $option['id'] ], $allowed_html ); // need to add slashes still before sending to the database
-                        $valid_input[ $option['id'] ] = addslashes( $input[ $option['id'] ] );
-                        break;
-                }
-                break;
```

Comment:
Nice cleanup! So we don't need all of these types anymore? They have not been used for a while it seems.

Reply:
Yes, not used and most probably not applicable to this module settings.
Other types are currently not used as well (e.g. `select`, `checkbox`, `multi-checkbox`), but maybe could be used in the future (validation logics to be reviewed in this case), so I left those there for now.
Additional cleanup could be needed when the module will be more mature and stable.